File: backend/nodejs/utils/crypto_hash.ts
import crypto from "crypto";
const SECRET = "ASDFSxzcbfdghdr345@@";
function crypto_hash(...args: any[]): string {
  let s = "";
  args.forEach((e) => {
    s = s + e.toString();
  });
  //console.log("args to string:", s);
  return crypto.createHmac("sha256", SECRET).update(s).digest("hex");
}

export { crypto_hash };
//console.log("hash is: ", crypto_hash("one", 2, [3]));

File: backend/golang/utils/crypto_hash.go
package utils

import (
	"crypto/sha256"
	"fmt"
)
func Crypto_hash(args ...string) (string, error) {
//func Crypto_hash(args ...[]byte) (string, error) {
	s := ""
	for _, i := range args {
		s += i
	}
	h := sha256.New()
	h.Write([]byte(s))
	return fmt.Sprintf("%x", h.Sum(nil)), nil
}